lib/AST/Expr.cpp

```diff
@@ -2211,6 +2211,20 @@ Expr *AutoClosureExpr::getUnwrappedCurryThunkExpr() const {
   return nullptr;
 }
 
+Expr *AutoClosureExpr::getUnwrappedCurryThunkExpr() const {
```

Comment:
@slavapestov Do you mean that `getUnwrappedCurryThunkCalledValue()` might be used instead of `getUnwrappedCurryThunkExpr()`? I don't think so: I intentionally moved the actual implementation into a separate `getUnwrappedCurryThunkImpl()` function, so we can either get an `Expr` or a `ValueDecl` depending of what we need.

`getUnwrappedCurryThunkExpr()` seems to have a number of uses over the code base, and that uses seem to expect `Expr` and not `ValueDecl`, so it's probably still needed. Anyway, even if some refactoring might be done (while I do not see a room for it in this particular case), probably it would be better to keep it separate from functional changes.

If I got your question wrong and this does not answer it, it would be great if you can clarify your point a bit.